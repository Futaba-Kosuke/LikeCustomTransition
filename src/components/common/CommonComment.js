/**
 * コメント表示用UIのベース部分を表示するコンポーネント
 */

import React from 'react';

//ReactNativeを使用したコンポーネントの呼び出し
import {
  StyleSheet,
} from 'react-native';

//NativeBaseを使用したコンポーネントの呼び出し
import {
  ListItem,
  Left,
  Body,
  Right,
  Text,
  Thumbnail
} from 'native-base';

//一覧シンプルリスト表示用のベースコンポーネントの内容を定義する
const CommonComment = ({ comment }) => {

  //取得した引数（オブジェクト：{ comment }）を分割する
  const { id, title, description, image_url, time } = comment;

  //表示する要素を返す
  return (
    <ListItem avatar>
      <Left>
        <Thumbnail source={image_url} />
      </Left>
      <Body>
        <Text>{title}</Text>
        <Text style={styles.commentTextStyle} note>{description}</Text>
      </Body>
      <Right>
        <Text note>{time}</Text>
      </Right>
    </ListItem>
  );
};

//このコンポーネントのスタイル設定
const styles = {
  commentTextStyle: {
    marginTop: 8,
    fontSize: 14,
    lineHeight: 18
  },
};

//このコンポーネントをインポート可能にする
export default CommonComment;
