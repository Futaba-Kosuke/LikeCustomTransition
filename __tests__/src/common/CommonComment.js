import 'react-native';
import React from 'react';
import CommonComment from '../../../src/components/common/CommonComment';
import { getCommentList } from '../../../src/stub/SampleDataStub';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('共通コンポーネント(コメント表示用部分)のスナップショットテスト', () => {
  const comment = getCommentList();
  const tree = renderer.create(
    <CommonComment comment={comment[0]} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
