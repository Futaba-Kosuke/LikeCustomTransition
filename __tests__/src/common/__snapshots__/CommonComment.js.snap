// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`共通コンポーネント(コメント表示用部分)のスナップショットテスト 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    avatar={true}
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": null,
        "borderColor": "#c9c9c9",
        "flexDirection": "row",
        "marginLeft": 15,
        "paddingBottom": null,
        "paddingRight": null,
        "paddingTop": null,
      }
    }
    virtual={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "alignSelf": "center",
          "flex": 0,
          "flexDirection": "row",
        }
      }
      virtual={undefined}
    >
      <Image
        source={1}
        style={
          Object {
            "borderRadius": 28,
            "height": 56,
            "width": 56,
          }
        }
        virtual={undefined}
      />
    </View>
    <View
      style={
        Object {
          "alignItems": null,
          "alignSelf": null,
          "borderBottomWidth": 0.5,
          "borderColor": "#c9c9c9",
          "flex": 1,
          "marginLeft": 15,
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
      virtual={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
            "marginLeft": null,
            "marginRight": 10,
          }
        }
        virtual={undefined}
      >
        ランチもディナーも大満足できる！
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        note={true}
        style={
          Object {
            "color": "#808080",
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "200",
            "lineHeight": 18,
            "marginLeft": null,
            "marginRight": 10,
            "marginTop": 8,
          }
        }
        virtual={undefined}
      >
        ランチもお手頃な値段で楽しむことができ、それでいてディナーも大満足のメニューラインナップ。また定期的にもお客さん参加型のイベント等も開催されておりとても楽しむことができます。
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "alignSelf": "stretch",
          "borderBottomWidth": 0.5,
          "borderColor": "#c9c9c9",
          "flex": 0,
          "paddingBottom": 10,
          "paddingLeft": null,
          "paddingRight": 15,
          "paddingTop": 10,
        }
      }
      virtual={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        note={true}
        style={
          Object {
            "alignSelf": null,
            "color": "#808080",
            "fontFamily": "System",
            "fontSize": 12,
            "fontWeight": "200",
          }
        }
        virtual={undefined}
      >
        3 hours ago
      </Text>
    </View>
  </View>
</View>
`;
